mne_bids/utils.py

```diff
@@ -26,9 +27,175 @@
+    acquisition: str | None
+        The acquisition parameters for the item. Corresponds to "acq".
+    run : int | None
+        The run number for this item. Corresponds to "run".
+    processing : str | None
+        The processing label for this item. Corresponds to "proc".
+    recording : str | None
+        The recording name for this item. Corresponds to "recording".
+    space : str | None
+        The coordinate space for an anatomical file. Corresponds to "space".
+    prefix : str | None
+        The prefix for the filename to be created. E.g., a path to the folder
+        in which you wish to create a file with this name.
+    suffix : str | None
+        The suffix for the filename to be created. E.g., 'audio.wav'.
+
+    Examples
+    --------
+    >>> bids_basename = make_bids_basename(subject='test', session='two',
+                                           task='mytask', suffix='data.csv')
+    >>> print(bids_basename)
+    sub-test_ses-two_task-mytask_data.csv
+    >>> print(bids_basename.__repr__())
+    BIDSPath (sub-test_ses-two_task-mytask_data.csv)
+    >>> # copy and update multiple entities at once
+    >>> new_basename = bids_basename.copy().update(subject='test2',
+                                                   session='one')
+    >>> print(new_basename)
+    sub-test2_ses-one_task-mytask_data.csv
+    """
+
+    def __init__(self, subject=None, session=None,
+                 task=None, acquisition=None, run=None, processing=None,
+                 recording=None, space=None, prefix=None, suffix=None):
+        self.update(subject=subject, session=session, task=task,
+                    acquisition=acquisition, run=run, processing=processing,
+                    recording=recording, space=space, prefix=prefix,
+                    suffix=suffix)
+
+    @property
+    def entities(self):
+        """Return dictionary of the BIDS entities."""
+        # create an ordered dictionary of all the bids entities
+        entities = OrderedDict()
+        for key in BIDS_PATH_ENTITIES:
+            value = getattr(self, key, None)
+            if value is not None:
+                entities[key] = value
+        return entities
+
+    def __str__(self):
+        """Return the string representation of the path."""
+        basename = _gen_bids_basename(**self.entities)
+        return basename
+
+    def __repr__(self):
+        """Representation in the style of `pathlib.Path`."""
+        return f"{self.__class__.__name__}({str(self)}) | \n" \
+               f"prefix={self.prefix}, \n" \
+               f"subject={self.subject}, \n" \
+               f"session={self.session}, \n" \
+               f"task={self.task}, \n" \
+               f"run={self.run}, \n" \
+               f"acquisition={self.acquisition}, \n" \
+               f"processing={self.processing}, \n" \
+               f"recording={self.recording}, \n" \
+               f"space={self.space}, \n" \
+               f"suffix={self.suffix}"
```

Comment:
How about:

```py
'\n'.join(repr(x).split(', '))[1:-1]
```

or something of that sort. Keeping the class name in there.

Reply:
Sorry I am confused with what you mean. This would give me an infinite recursion. Is x supposed to be the various BIDS Path entities?

Reply:
sorry I was playing around with it in my terminal. Better now?

Reply:
Let me know if the change is what you meant... It ends up looking like this in terminal:

```
bids_basename = BIDSPath(sub-01_ses-02_acq-03_run-2) | 
subject=01
session=02
acquisition=03
run=2
```